spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java

```diff
@@ -73,12 +73,14 @@
 		@ConditionalOnMissingBean
 		public ServiceInstanceListSupplier discoveryClientServiceInstanceListSupplier(
 				ReactiveDiscoveryClient discoveryClient, Environment env,
-				ObjectProvider<CacheManager> cacheManager) {
+				ApplicationContext context) {
 			DiscoveryClientServiceInstanceListSupplier delegate = new DiscoveryClientServiceInstanceListSupplier(
 					discoveryClient, env);
-			if (cacheManager.getIfAvailable() != null) {
+			if (context.getBeanProvider(LoadBalancerCacheManager.class)
```

Comment:
is it more efficient to call `context.getBeanProvider` once?  regardless it would make the code a bit cleaner.